### PR TITLE
Cilium kube proxy migration hacks

### PR DESCRIPTION
**Pull request by whites11**
This PR introduces another bunch of hacks needed to smoothly migrate between aws-cni and cilium with kube proxy replacement.

More specifically, this PR:
- manages an annotation in the Cluster CR in order to enable or disable cilium's kube-proxy feature during upgrade. Timing is very important so aws-operator has the controller role in this process.
- adds a sidecar container to aws-node pods that adjusts the node's routing tables in order for aws-cni pods to be able to connect to cilium pods during migration

Sorry this is a big PR, but there is no meaningful way to split it.

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by T-Kukawka** [comment]
i will leave this review for Nick

File: service/controller/key/cilium.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+import (
+	"github.com/giantswarm/k8smetadata/pkg/annotation"
+	apiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+func ForceDisableCiliumKubeProxyReplacement(cluster apiv1beta1.Cluster) bool {
+	v, found := cluster.Annotations[annotation.CiliumForceDisableKubeProxyAnnotation]
+
+	return found && v == "true"
```
Comment (njuettner) [inline service/controller/key/cilium.go:11]: Do we wanna check the value or would it be enough to just look if the annotation is set?

**Review by njuettner — APPROVED** [review]
lgtm

File: service/controller/key/cilium.go (hunk 1 above)
Comment (whites11) [reply service/controller/key/cilium.go:11]: I would be confused if I see the annotation set to "false" but the behaviour still happening.
I know elsewhere we only check the presence of the annotation, but I think it's best to be explicit and check the value as well.

Comment (njuettner) [reply service/controller/key/cilium.go:11]: Makes sense 👍🏻

**Pull request merged**